src/lib.rs

```diff
@@ -398,6 +398,19 @@ impl EndpointHeader {
         let data = unsafe { access.read(self.0, 0x3c) };
         (data.get_bits(8..16) as u8, data.get_bits(0..8) as u8)
     }
+
+    pub fn update_interrupt<F>(&mut self, access: &impl ConfigRegionAccess, f: F)
```

Comment:
Why does this require &mut self, rather than &self like update_command?

Reply:
Good point!

I just scrolled to the next similar function, `update_interrupt`: https://github.com/rust-osdev/pci_types/blob/07959ecb9d541221ca9a1fc1d3b21e320b85ce3c/src/lib.rs#L438

`write_bar` also takes `&mut self`. I am not sure which one we'd want to settle on but there's some consistency to gain here for sure. 🤔

Reply:
Hm yeah some consistency here would be sensible. Obviously they can be `&self` because writes to PCI space can be, but I then usually make things `&mut` if they 'feel like' mutation. This isn't very scientific, but I think helps avoid bugs by enforcing borrow checking at your higher levels of abstraction.

I'd accept a PR making these methods consistent either way tbh, if one of you feels strongly one one or the other.

Reply:
I don't feel strongly about this either, but would also lean towards `&mut`. @bjorn3, how do you feel about this?

Reply:
I don't have a strong preference either way. Just that it should be consistent everywhere.

Reply:
Should we change `ConfigRegionAccess::write(&self, address: PciAddress, offset: u16, value: u32)` too? 🤔

Reply:
I don't think we should. Several PCI functions can be reconfigured at the same time just fine and `ConfigRegionAccess` is shared between all PCI functions within a single PCI domain.

Reply:
I opened https://github.com/rust-osdev/pci_types/pull/32.